R/load_austraits.R

```diff
@@ -125,16 +125,23 @@ create_metadata <- function(res){
 #' @param path file path to where AusTraits will be downloaded
 
 download_austraits <- function(url, filename, path) {
+  # Get user timeout option
+  o <- getOption('timeout')
+  
+  # Set max timeout
+  options(timeout = max(300, getOption("timeout")))  
+  
   #Download latest build
   fn <- paste(tempfile(), '.download', sep='') #Temporary folder
   
   message("Downloading AusTraits to '", path,"'")
   
   res <- utils::download.file(url=url, destfile=fn, method="auto", quiet = FALSE, mode = "wb", cacheOK = TRUE)
   
-  if (res == 0) { #Warning hygiene
+  if (res == 0) { #Warning and timeout hygiene
     w <- getOption('warn') #save option
-    on.exit(options('warn' = w)) #set it back
+    on.exit(options(warn = w,
```

Comment:
Nice! I think it's better to move this call up to line 130, right before you change the options. Thatway it will reinstate the options, even if the function fails (errors) during download.